en/setup/dev_env_linux.md

```diff
@@ -14,63 +37,113 @@ sudo usermod -a -G dialout $USER
+
+<div class="host-code"></div>
+
+```sh
+mkdir -p $HOME/ninja
+cd $HOME/ninja
+wget https://github.com/martine/ninja/releases/download/v1.6.0/ninja-linux.zip
+unzip ninja-linux.zip
+rm ninja-linux.zip
+exportline="export PATH=$HOME/ninja:\$PATH"
+if grep -Fxq "$exportline" ~/.profile; then echo nothing to do ; else echo $exportline >> ~/.profile; fi
+. ~/.profile
+```
+
+## Common Dependencies
+
+Update the package list and install the following dependencies for all PX4 build targets. 
+
+> **Info** Install the [Ninja Build System](#ninja-build-system) for faster build times than with `Make`. It will be automatically selected if installed.
 
 ```sh
 sudo add-apt-repository ppa:george-edison55/cmake-3.x -y
 sudo apt-get update
 sudo apt-get install python-argparse git-core wget zip \
     python-empy qtcreator cmake build-essential genromfs -y
-# simulation tools
-sudo apt-get install ant protobuf-compiler libeigen3-dev libopencv-dev openjdk-8-jdk openjdk-8-jre clang-3.5 lldb-3.5 -y
 # required python packages
 sudo apt-get install python-pip
 sudo -H pip install pandas jinja2
 ```
 
-### NuttX based hardware
+## Simulation Dependencies
+The dependencies for the Gazebo and jMAVSim simulators listed below. You should minimally install jMAVSim to make it easy to test the installation. Additional information about these and other supported simulators is covered in: [Simulation](../simulation/README.md).
 
-Ubuntu comes with a serial modem manager which interferes heavily with any robotics related use of a serial port \(or USB serial\). It can deinstalled without side effects:
+### jMAVSim
 
-```sh
-sudo apt-get remove modemmanager
+Install the dependencies for [jMAVSim Simulation](../simulation/sitl.md).
+
+```
+# jMAVSim simulator
+sudo apt-get install ant openjdk-8-jdk openjdk-8-jre -y
 ```
 
-Update the package list and install the following dependencies. Packages with specified versions should be installed with this particular package version.
+### Gazebo
+
+Install the dependencies for [Gazebo Simulation](../simulation/gazebo.md).
+
+```
+# Gazebo simulator
+sudo apt-get install protobuf-compiler libeigen3-dev libopencv-dev -y
+sudo sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list'
+## Setup keys
+wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
+## Update the debian database:
+sudo apt-get update -y
+## Install Gazebo8
+sudo apt-get install gazebo8 -y
+## For developers (who work on top of Gazebo) one extra package
+sudo apt-get install libgazebo8-dev
+```
+
+<!-- these dependencies left over when I separated the dependencies. These appear to both be for using Clang. MOve them down?
+sudo apt-get install clang-3.5 lldb-3.5 -y
```

Comment:
@LorenzMeier clang-3.5 lldb-3.5 were listed in simulator dependencies, but certainly are not needed for Gazebo or jMAVSim. I think perhaps are RaPi dependencies. 
Problem is, that the RaPi section suggests going to a repo to "find best option". Should these be installed/used or moved to RaPi?